### DISCUSSION TITLE
Static Table Data Sources

### DISCUSSION THREAD

**Discussion by flavio-schoute** [filamentphp/filament#14099] [General]
I was reading this article on the [filament php website.](https://filamentphp.com/content/alexandersix-filament-what-to-expect-in-2024#static-table-data-sources)

Until I saw this:

`Currently, Filament's tables require Eloquent to populate data. Not everything worth putting in a table is retrievable with Eloquent, though! In 2024, we want to do some work on Filament's tables to allow them to populate data from an array or from a function returning an array. This would dramatically increase the usefulness of Filament tables across all parts of an application, not just those parts that involve models in a database.`

My question is what is expected date for this future or when does V4 come out?

**Comment by leandrocfe** [comment]
Probably available on V4

**Comment by e2d2-dev** [comment]
just use sushi:

[https://github.com/calebporzio/sushi](url)

**Reply by blackchineykh** [reply]
How can we use sushi to fetch data from an api with pagination?

**Reply by micraux** [reply]
Correct link to sushi: [https://github.com/calebporzio/sushi](https://github.com/calebporzio/sushi)

**Comment by indigoram89** [comment]
@danharrin can we hope for this opportunity to be realized?

**Comment by mc-fu** [comment]
just found the [link](https://filamentphp.com/content/alexandersix-filament-what-to-expect-in-2024#static-table-data-sources) again

any news on this? it would be so great to see this feature in v4

**Comment by danharrin** [comment]
Yes its on v4

**Reply by mc-fu** [reply]
great news, thank you so much!

**Reply by blackchineykh** [reply]
Is v4 ready for use?

**Reply by danharrin** [reply]
No

**Reply by blackchineykh** [reply]
Thanks. Will pagination using external api sources also be supported in v4?

**Reply by danharrin** [reply]
Yes

**Reply by indigoram89** [reply]
wow! it is very great news! thank you so much
